src/chuchichaestli/models/unet/unet.py

```
@@ -59,115 +68,200 @@
+              to all blocks in a level, otherwise only to the first block in a level.
+            add_noise: Add a Gaussian noise regularizer block in the bottleneck (before or after).
+              Can be "up" (after the bottlenet) or "down" (before the bottleneck).
+            noise_sigma: Std. relative (to the magnitude of the input) for the noise generation.
+            noise_detached: If True, the input is detached for the noise generation.
+              Note, this should generally be `True`, otherwise the noise is learnable.
         """
         super().__init__()
 
-        if dimensions not in DIM_TO_CONV_MAP:
-            raise ValueError(
-                f"Invalid number of dimensions ({dimensions}). Must be one of {list(DIM_TO_CONV_MAP.keys())}."
-            )
-
-        if len(down_block_types) != len(up_block_types):
-            raise ValueError("The number of down and up block types must be equal.")
-
-        if len(down_block_types) != len(block_out_channel_mults):
-            raise ValueError(
-                "The number of down block types and output channel multipliers must be equal."
-            )
+        self._validate_inputs(
+            dimensions, down_block_types, up_block_types, block_out_channel_mults
+        )
 
+        # Cache commonly used values
         conv_cls = DIM_TO_CONV_MAP[dimensions]
+        upsample_cls = UPSAMPLE_FUNCTIONS[upsample_type]
+        downsample_cls = DOWNSAMPLE_FUNCTIONS[downsample_type]
+        n_mults = len(block_out_channel_mults)
+        self.num_blocks_per_level = num_blocks_per_level
+        self.skip_connection_to_all_blocks = skip_connection_to_all_blocks
 
+        # Group normalization optimization
         if res_norm_type == "group" and n_channels % res_groups != 0:
             warnings.warn(
                 f"Number of channels ({n_channels}) is not divisible by the number of groups ({res_groups}). Setting number of groups to in_channels."
```

We could raise an error instead of the warning...?

---

This  is a decision torch made. The group normalization class automatically clips the groups if the number of channels is smaller than the number of groups. I think warnings are better here, because it just points out that your setting has no effect.